File: 2025-1/sesion-de-clase-24/java/src/main/java/Deposito.java
/**
 *
 * @author eric
 */
public class Deposito implements Runnable {
    private final double monto;
    private final CuentaBancaria cuenta;
    
    public Deposito(CuentaBancaria cuenta, double monto) {
        this.monto = monto;
        this.cuenta = cuenta;
    }
    
    @Override
    public void run() {
        String nombreCliente = Thread.currentThread().getName();
        this.cuenta.depositar(nombreCliente, monto);
    }
}

File: 2026-1/sesion-de-clase-24/net/sesion-de-clase-24/Ejercicio4/Deposito.cs
using System.Threading;

namespace Ejercicio4;

public class Deposito
{
    public CuentaBancaria Cuenta { get; set; }

    public double Monto { get; set; }

    public Deposito(CuentaBancaria cuenta, double monto)
    {
        Cuenta = cuenta;
        Monto = monto;
    }

    public void Run()
    {
        string nombreCliente = Thread.CurrentThread.Name ?? "Sin nombre";
        Cuenta.Depositar(nombreCliente, Monto);
    }
}
